foldamers/cg_model/cgmodel.py

```diff
@@ -578,10 +416,10 @@ def get_bond_list(self):
                         int(monomer) != 0
                         and backbone_bead == 0
                         and monomer_type["backbone_length"] - 1
-                        in [monomer_type["sidechain_positions"]]
+                        in monomer_type["sidechain_positions"]
                     ) or (
                         backbone_bead != 0
-                        and backbone_bead - 1 in [monomer_type["sidechain_positions"]]
+                        and (backbone_bead - 1) in monomer_type["sidechain_positions"]
                     ):
                         parent_index = bead_index - monomer_type["sidechain_length"] - 1
                     else:
```

Comment:
I think the problem is that it's bonding the backbone beads based on sidechain length of the _current_ monomer type, not the previous which can now have different numbers of sidechain beads.

At any rate we should revisit this logic - it's a bit hard to follow even for homopolymers.

Reply:
Good detective work.  Let me try a patch for this.  Let's wait to reorganize the logic until we change it up so we can have arbitrary topology monomers / particle types, since we want to do that relatively soon.

Reply:
OK, I think this new commit should fix that.